compiler/ballerina-parser/src/test/resources/statements/assignment-stmt/assignment_stmt_source_15.bal

```diff
@@ -0,0 +1,3 @@
+function foo() {
+    [error, boolean...] [error(g, ), ...j] = [error("error!"), true];
```

Comment:
This is not an [assignment-stmt](https://ballerina.io/spec/lang/master/#assignment-stmt). This is a [local-var-decl-stmt](https://ballerina.io/spec/lang/master/#local-var-decl-stmt). Since the problem was in the binding pattern recovery shall we move those test cases to `TypedBindingPatternsTest`?

Reply:
Updated